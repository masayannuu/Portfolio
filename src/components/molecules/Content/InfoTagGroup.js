import React from 'react'
import PropTypes from 'prop-types'
import { Control, Field } from 'reactbulma'

import InfoTag from '../../atoms/Label/InfoTag'
import ParagraphTitle from '../../atoms/Paragraph/ParagraphTitle'

const InfoTagGroup = ({ name, tags }) => (
  <Field grouped>
    <ParagraphTitle>{name}</ParagraphTitle>
    <Field grouped className="is-grouped-multiline">
      {tags.map(tag => (
        <Control key={tag}>
          <InfoTag key={tag}>{tag}</InfoTag>
        </Control>
      ))}
    </Field>
  </Field>
)

InfoTagGroup.propTypes = {
  name: PropTypes.string.isRequired,
  tags: PropTypes.array.isRequired
}

export default InfoTagGroup
