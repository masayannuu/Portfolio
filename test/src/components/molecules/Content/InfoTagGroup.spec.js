import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'

import InfoTagGroup from '../../../../../src/components/molecules/Content/InfoTagGroup'

Enzyme.configure({ adapter: new Adapter() })

const tags = ['Ruby', 'Js', 'java', 'C#']
describe('Content InfoTagGroup Molecules', () => {
  it('should render InfoTagGroup', () => {
    const wrapper = Enzyme.shallow(<InfoTagGroup name="Language" tags={tags} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
