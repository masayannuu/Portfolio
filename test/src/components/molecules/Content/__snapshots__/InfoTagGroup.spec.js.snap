// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content InfoTagGroup Molecules should render InfoTagGroup 1`] = `
<Field
  className="is-grouped-multiline"
  grouped={true}
>
  <InfoTagGroup__StyledTitle>
    Language
  </InfoTagGroup__StyledTitle>
  <InfoTagGroup__TagGroupField
    className="is-grouped-multiline"
    grouped={true}
  >
    <Control
      as="div"
      key="Ruby"
    >
      <InfoTag
        key="Ruby"
      >
        Ruby
      </InfoTag>
    </Control>
    <Control
      as="div"
      key="Js"
    >
      <InfoTag
        key="Js"
      >
        Js
      </InfoTag>
    </Control>
    <Control
      as="div"
      key="java"
    >
      <InfoTag
        key="java"
      >
        java
      </InfoTag>
    </Control>
    <Control
      as="div"
      key="C#"
    >
      <InfoTag
        key="C#"
      >
        C#
      </InfoTag>
    </Control>
  </InfoTagGroup__TagGroupField>
</Field>
`;
